notion.py

```diff
@@ -90,3 +99,186 @@ def group_by_user_raw_data(raw_data: list) -> dict:
             }
             users_group_data.update(user_dist_data)
     return users_group_data
+
+
+def get_last_duties_by_cohort(cohort: Cohort, count_days: int = 7) -> dict:
+    """
+    Получить последовательность из `:count` крайних дежурств в обратном порядке.
+    Данные из прошлого не учитываем.
+    :param cohort: когорта,
+    :param count_days: количество дней для расчета,
+    :return:
+        {
+            'last_mentors_ids': tuple, крайние дежурства. **Порядок обратный --** от последнего к первому.
+            'actual_date': date(%Y-%m-%d), крайняя заполненная дата (%Y-%m-%d)
+        }
+    """
+    date_now = datetime.date.today().__str__()
+    databases_duties = query_databases_duties(
+        database_id=cohort.notion_db_id,
+        date=date_now,
+        page_size=count_days,
+    )
+
+    if len(databases_duties) < 2:
+        logging.error('Недостаточно данных для расчета расписания. Нужен минимум один цикл дежурств в будущем.')
+        return False
+    last_duties = [duty.properties for duty in databases_duties]
+    last_duties = sorted(last_duties, key=lambda duty: duty.Дата.date.start, reverse=True)
+    actual_date = parse(last_duties[0].Дата.date.start).date()
```

Comment:
Можно не подключать новую библиотеку, а сделать всё средствами стандартного питона
```suggestion
    actual_date = datetime.datetime.strptime(last_duties[0].Дата.date.start, '%Y-%m-%d').date()
```
Подключать новые библиотеки всегда плохо, если задачу можно решить без их

Reply:
тут я пытался предусмотерть случай, когда в календате когорты поле `Дата` укажут не совсем в таком формате как мы ожидаем. 
Как я понял, в ноушене это очень вероятно. 
Но это не точно )